test/screenshot.test.js

```diff
@@ -0,0 +1,130 @@
+const assert = require('assert');
+const mkdirp = require('mkdirp');
+const { PNG } = require('pngjs');
+const match = require('pixelmatch');
+const YAML = require('yaml');
+
+const { writeFile } = fsPromises;
+
+const ACKNOWLEDGED_COMMIT = '247b7e6815b358d7962ad2bc21d34dd9122bbe69';
+const LOCAL_HOST = `http://localhost:${process.env.JEST_PORT}`;
+const REMOTE_HOST = 'https://design.login.gov';
+const DIFF_DIRECTORY = join(__dirname, '../tmp/results/screenshot-diff');
+const { url: URL_PREFIX, github_repo_url: GITHUB_REPO_URL } = YAML.parse(
+  readFileSync(join(__dirname, '../_config.yml'), 'utf8'),
+);
+
+async function getURLsFromSitemap(url) {
+  await page.goto(url);
+  return page.$$eval('url loc', locs => locs.map(loc => loc.textContent));
+}
+
+async function stubAnimations() {
+  await page.evaluate(() => {
+    const isGif = img => img.src.endsWith('.gif');
+    function stubGif(img) {
+      img.width = img.width;
+      img.height = img.height;
+      img.src = 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==';
+      img.removeAttribute('srcset');
+    }
+
+    [...document.querySelectorAll('img')].filter(isGif).forEach(stubGif);
+  });
+}
+
+function getLatestCommit() {
+  const [sha] = spawnSync('git', ['ls-remote', GITHUB_REPO_URL, 'HEAD'])
+    .stdout
+    .toString()
+    .trim()
+    .split('\t');
+
+  return sha;
+}
+
+async function getScreenshot(url) {
+  await page.goto(url);
+  await stubAnimations();
+  return page.screenshot({ fullPage: true });
+}
+
+function getURLPath(url) {
+  const prefix = new URL(URL_PREFIX).pathname;
+  const { pathname } = new URL(url);
+  return pathname.indexOf(prefix) === 0 ? pathname.slice(prefix.length) : pathname;
+}
+
+function getDiffOutputBaseFileName(pathname) {
+  let normalizedPathname = pathname;
+  normalizedPathname = normalizedPathname.replace(/^\/|\/$/g, '');
+  normalizedPathname = normalizedPathname.replace(/\W/g, '_');
+  normalizedPathname = normalizedPathname || 'home';
+
+  return join(DIFF_DIRECTORY, normalizedPathname);
+}
+
+function fillImageToSize(image, width, height) {
+  if (image.width === width && image.height === height) {
+    return image;
+  }
+
+  const resizedImage = new PNG({ width, height, fill: true });
+  PNG.bitblt(image, resizedImage, 0, 0, image.width, image.height, 0, 0);
+
+  for (let y = image.height; y < height; y += 1) {
+    for (let x = image.width; x < width; x += 1) {
+      // eslint-disable-next-line no-bitwise
+      const index = ((resizedImage.width * y) + x) << 2;
+      resizedImage.data[index] = 0;
+      resizedImage.data[index + 1] = 0;
+      resizedImage.data[index + 2] = 0;
+      resizedImage.data[index + 3] = 64;
+    }
+  }
+
+  return resizedImage;
+}
+
+const testFn = getLatestCommit() === ACKNOWLEDGED_COMMIT ? test.skip : test;
+
+testFn('screenshot visual regression', async () => {
+  const paths = (await getURLsFromSitemap(`${LOCAL_HOST}/sitemap.xml`)).map(getURLPath);
```

Comment:
WDYT of using the remote sitemap? because if we add a new page, it'll only exist locally and won't exist remote, so it'll automatically fail right?

Reply:
> WDYT of using the remote sitemap? because if we add a new page, it'll only exist locally and won't exist remote, so it'll automatically fail right?

Yeah, that makes sense to me.

Originally I'd noticed the URLs would be the simple paths when in development mode, which (combined with shorter turnaround time) was the primary reason for using the local URL. I later discovered the pre-test script will build for production, so we need to trim the [prefix URL](https://github.com/18F/identity-style-guide/blob/086d240b4bad78f9e7bc892ee7d907fe9d085a96/_config.yml#L4) anyways. Swapping to the remote URL would be trivial then, and avoid those issues you describe.

Reply:
> WDYT of using the remote sitemap? because if we add a new page, it'll only exist locally and won't exist remote, so it'll automatically fail right?

Updated in 1cddbd0.